instrumentation/kamon-pekko-http/src/test/scala/kamon/pekko/http/PekkoHttpServerTracingSpec.scala

```diff
@@ -0,0 +1,280 @@
+            span.operationName shouldBe expected
+          }
+        }
+
+        "including ambiguous nested directives" in {
+          val path = s"v3/user/3/post/3"
+          val expected = "/v3/user/{}/post/{}"
+          val target = s"$protocol://$interface:$port/$path"
+          client.newCall(new Request.Builder().url(target).build()).execute()
+
+          eventually(timeout(10 seconds)) {
+            val span = testSpanReporter().nextSpan().value
+            span.operationName shouldBe expected
+          }
+        }
+      }
+
+      "change the Span operation name when using the operationName directive" in {
+        val target = s"$protocol://$interface:$port/$traceOk"
+        client.newCall(new Request.Builder().url(target).build()).execute()
+
+        eventually(timeout(10 seconds)) {
+          val span = testSpanReporter().nextSpan().value
+          span.operationName shouldBe "user-supplied-operation"
+          span.tags.get(plain("http.url")) should endWith(s"$interface:$port/$traceOk")
+          span.metricTags.get(plain("component")) shouldBe "pekko.http.server"
+          span.metricTags.get(plain("http.method")) shouldBe "GET"
+          span.metricTags.get(plainLong("http.status_code")) shouldBe 200L
+        }
+      }
+
+      "mark spans as failed when request fails" in {
+        val target = s"$protocol://$interface:$port/$dummyPathError"
+        client.newCall(new Request.Builder().url(target).build()).execute()
+
+        eventually(timeout(10 seconds)) {
+          val span = testSpanReporter().nextSpan().value
+          span.operationName shouldBe s"/$dummyPathError"
+          span.tags.get(plain("http.url")) should endWith(s"$interface:$port/$dummyPathError")
+          span.metricTags.get(plain("component")) shouldBe "pekko.http.server"
+          span.metricTags.get(plain("http.method")) shouldBe "GET"
+          span.metricTags.get(plainBoolean("error")) shouldBe true
+          span.metricTags.get(plainLong("http.status_code")) shouldBe 500L
+        }
+      }
+
+      "change the operation name to 'unhandled' when the response status code is 404" in {
+        val target = s"$protocol://$interface:$port/unknown-path"
+        client.newCall(new Request.Builder().url(target).build()).execute()
+
+        eventually(timeout(10 seconds)) {
+          val span = testSpanReporter().nextSpan().value
+          span.operationName shouldBe "unhandled"
+          span.tags.get(plain("http.url"))  should endWith(s"$interface:$port/unknown-path")
+          span.metricTags.get(plain("component")) shouldBe "pekko.http.server"
+          span.metricTags.get(plain("http.method")) shouldBe "GET"
+          span.metricTags.get(plainBoolean("error")) shouldBe false
+          span.metricTags.get(plainLong("http.status_code")) shouldBe 404L
+        }
+      }
+
+      "correctly time entity transfer timings" in {
+        val target = s"$protocol://$interface:$port/$stream"
+
+        try {
```

Comment:
Interesting finding here, the akka-stream dependency is 'hard-coded' to 2.5.32 in the kamon-akka-http module. When this dependency is updated to 2.6.20 the test fails in the same way. the first server (HTTP) gets an exception. However the metrics etc are already recorded in Kamon, hence a green test when the exception handling is a no-op. 


```
java.net.SocketException: Connection reset
  | at java.net.SocketInputStream.read(SocketInputStream.java:210)
	at java.net.SocketInputStream.read(SocketInputStream.java:141)
	at okio.InputStreamSource.read(JvmOkio.kt:94)
	at okio.AsyncTimeout$source$1.read(AsyncTimeout.kt:125)
	at okio.RealBufferedSource.indexOf(RealBufferedSource.kt:427)
	at okio.RealBufferedSource.readUtf8LineStrict(RealBufferedSource.kt:320)
	at okhttp3.internal.http1.HeadersReader.readLine(HeadersReader.kt:29)
	at okhttp3.internal.http1.Http1ExchangeCodec.readResponseHeaders(Http1ExchangeCodec.kt:178)
	at okhttp3.internal.connection.Exchange.readResponseHeaders(Exchange.kt:106)
	at okhttp3.internal.http.CallServerInterceptor.intercept(CallServerInterceptor.kt:79)
	at okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.kt:109)
	at okhttp3.internal.connection.ConnectInterceptor.intercept(ConnectInterceptor.kt:34)
	at okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.kt:109)
	at okhttp3.internal.cache.CacheInterceptor.intercept(CacheInterceptor.kt:95)
	at okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.kt:109)
	at okhttp3.internal.http.BridgeInterceptor.intercept(BridgeInterceptor.kt:83)
	at okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.kt:109)
	at okhttp3.internal.http.RetryAndFollowUpInterceptor.intercept(RetryAndFollowUpInterceptor.kt:76)
	at okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.kt:109)
	at okhttp3.internal.connection.RealCall.getResponseWithInterceptorChain$okhttp(RealCall.kt:201)
	at okhttp3.internal.connection.RealCall.execute(RealCall.kt:154)
	at kamon.pekko.http.PekkoHttpServerTracingSpec.$anonfun$testSuite$27(PekkoHttpServerTracingSpec.scala:233)